lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
> Thing is, I want to avoid extra filesystem calls if I can help it.

Yeah I totally understand that, but that said I don't think neither the tree-view nor tabs should be the sources of truth about that kind of information, and enforcing this pattern even we see its flaws right now seems the wrong thing to do. Atom could be where we query and process file stats, but we don't have this abstraction yet unfortunately.

Did you benchmark how file icons performs when issuing fs calls? How better it performs after putting a cache behind it? Maybe you could warm the cache in the background on a separate process? Alternatively, since this is an IO-bound operation, we might think to make the method asynchronous and apply the classes to the DOM node when the operation completes.